Tribler/community/tunnel/hidden_community.py

```diff
@@ -312,26 +397,34 @@ def on_key_request(self, messages):
 
     def check_key_response(self, messages):
         for message in messages:
-            if not message.source.startswith(u"circuit_"):
```

Comment:
Exactly, but not over the originating circuit, causing connectability problems when the downloader is not connectable